# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [3]:
import numpy as np
from scipy import stats

1.Гіпотези

H₀: μ = 123.7 (середній час від першого заходу на сайт до покупки)
H₁: μ > 123.7 (середній час збільшився)
Тип тесту: правосторонній

In [36]:
#2
n = 7
mu0 =123.7
alpha = 0.10

time = np.array([128, 135, 121, 142, 126, 151, 123])
x = np.mean(time)
s = np.std(time, ddof=1) 
print(f"Вибіркове середнє: {x:.2f}")
print(f"Вибіркове стандартне відхилення: {s:.2f}")

Вибіркове середнє: 132.29
Вибіркове стандартне відхилення: 10.98


In [37]:
#3
t_stat = (x - mu0) / (s / np.sqrt(n))

df = n - 1
t_crit = stats.t.ppf(1 - alpha, df)

p_value = 1 - stats.t.cdf(t_stat, df)

print(f"t-статистика: {t_stat:.3f}")
print(f"Критичне значення (df={df}): {t_crit:.3f}")
print(f"p-value: {p_value:.4f}")

t-статистика: 2.069
Критичне значення (df=6): 1.440
p-value: 0.0420


In [39]:
#4
if p_value < alpha:
    print(f"Відхиляємо H0: середній час від першого заходу на сайт до покупки більший за 123.7 хв на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Відхиляємо H0: середній час від першого заходу на сайт до покупки більший за 123.7 хв на рівні значущості 0.1


5. На рівні значущості α = 0.10 є статистично значущі докази того, що редизайн сайту призвів до збільшення середнього часу до покупки.

In [41]:
alpha = 0.05
if p_value < alpha:
    print(f"Відхиляємо H0: середній час від першого заходу на сайт до покупки більший за 123.7 хв на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Відхиляємо H0: середній час від першого заходу на сайт до покупки більший за 123.7 хв на рівні значущості 0.05


6.Навіть при рівні значущості α = 0.05 ми відхиляємо нульову гіпотезу, оскільки p-value менше за α. Отже, висновок про збільшення часу до покупки залишається незмінним.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [43]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count = 311   # кількість "успіхів"
nobs = 1200   # обсяг вибірки
p0 = 0.23      # гіпотетична пропорція (H0)

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.05, method='normal')

print("Тест для однієї пропорції")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"97% довірчий інтервал: {confint}")

Тест для однієї пропорції
Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.23437491020923482, 0.28395842312409847)


H₀: p = 23 
H₁: p > 23
Тип тесту: правосторонній

Оскільки p-value = 0.0106 < 0.03, відхиляємо H₀
      Є статистично значущі докази, що частка тих, хто упізнавали логотип компанії, більша за 23 %

97% довірчий інтервал становить (0.234; 0.284), тобто справжня частка впізнаваності, ймовірно, знаходиться в межах 23.4%–28.4%.
Оскільки весь інтервал лежить вище 23%, це додатково підтверджує висновок про зростання впізнаваності логотипу.
